### PR TITLE
added change log for release 0.24.0

### PR DESCRIPTION
**Pull request by sohailc**
Release branch for 0.24.0

### CODE REVIEW

File: docs/changes/0.24.0.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+- support export to xarray for data with non unique indices (#2848).
+- update use of Slack API in qcodes.utils.slack.Slack class (#2813)
+
+
+-----------------
+Improved drivers:
+-----------------
+
+- Remove hardcoded acquire mode and allow users to set acquire mode for trace on Infiniium driver (#2837)
```
Comment (jenshnielsen) [inline docs/changes/0.24.0.rst:18]: As I wrote in #2837 this should be documented as a breaking change

File: docs/changes/0.24.0.rst [hunk 2]
```diff
@@ -0,0 +1,19 @@
+Changelog for QCoDeS 0.24.0
+===========================
+
+The March 2021 release of QCoDeS.
+
```
Comment (jenshnielsen) [inline docs/changes/0.24.0.rst:5]: https://github.com/QCoDeS/Qcodes/pull/2782 is missing
 
 
 https://github.com/QCoDeS/Qcodes/pull/2854 should also land before the release

File: docs/changes/0.24.0.rst [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+- Update use of Slack API in qcodes.utils.slack.Slack class (#2813)
+- Support dataset export to NetCDF or CSV after measurement ends (#2782)
+- Export more metadata to xarray dataset (#2854)
+
+
+-----------------
+Improved drivers:
+-----------------
+
+- Remove hardcoded acquire mode and allow users to set acquire mode for trace on Infiniium driver (#2837) *Note: Because the mode of the instrument is no longer set in the constructor, this is a breaking change.*
```
Comment (astafan8) [inline docs/changes/0.24.0.rst:20]: for breaking changes, if they are present, we usually add a separate section (see previous changelogs), this makes it more visible to the users.

**Review by astafan8 — APPROVED** [review]

File: docs/changes/0.24.0.rst (hunk 3 above)
Comment (astafan8) [reply docs/changes/0.24.0.rst]: yes, indeed, like that. however we usually put the breaking changes at the top for visibility :)

**Comment by FarBo** [comment]
@sohailc 
There is a last minute notebook update, which can be mentioned in the change log :)
Extra explanation for Experiment and database in 15_minutes notebook #2838

**Pull request merged**